src/widgetastic_patternfly/__init__.py

```diff
@@ -2800,7 +2800,10 @@ def type(self):
 class FlashMessages(View):
     """Represent the div block containing the individual inline notifications."""
     ROOT = './/div[@id="flash_msg_div"]'
-    MSG_LOCATOR = './div[contains(@class, "flash_text_div")]/div[contains(@class, "alert")]'
+    MSG_LOCATOR = (
+        '//div[contains(@class, "flash_text_div") or contains(@class, "flash-message")]'
```

Comment:
The xpath shouldn't be changed to be a non-relative locator. The syntax was `./div[...]` because its relative to the ROOT locator, and the methods within the class using this locator are doing it relative to `parent=self`.

Just adding ` or contains(@class, 'flash-message')` to the specification for the first div should be sufficient to match.

Reply:
What is the test case that is failing? Does this need to be a change to FlashMessages itself, or if it's an issue only with certain pages/views, could you instead have a subclass of FlashMessages with a custom MSG_LOCATOR for use in the specific views that have this problem?

Reply:
@tpapaioa issue is with some test cases but its more general solution. As per my personal observation @mshriver suggestion is right and it should not effect on current test cases/implementation. `ROOT` set your working area for specific widget (with browser instance) i.e. `flash_text_div` so class attribute `flash-message` web element should work without any trouble.

Reply:
Which tests/views have this issue? Why was `div[contains(@class, "alert")]` removed and the relative `'./div'` changed to `'//div'` at the start of `MSG_LOCATOR`?

Reply:
@tpapaioa double check your diff view, `div[contains(@class, "alert")]` is still part of the locator

Reply:
Also note this case is a _little_ screwy in the first place, because flash message is an MIQ thing, Patternfly3 just has the alert divs for `Inline Notifications` widget.

So 'FlashMessage' in wt.pf should really be `InlineNotification`, and widgetastic_manageiq module within integration_tests should define `FlashMessages` parent.  The message locator here is really relevant to widgetastic_manageiq, as this isn't a Patternfly class.

That being said I don't want this PR to turn into removing FlashMessages from the library, which is why I'm open to accepting more classes for the div xpath for individual messages. Anything other than the alert class isn't strictly Patternfly3 anyway.

Reply:
I see the `alert` class in the diff now. I still think the relative locator should be kept at the beginning of the locator, and it would be good to know what views have this issue. I don't see 'flash-message' in any of the manageiq ui repositories....

Reply:
> Which tests/views have this issue?

@tpapaioa this test - [test_upload_blank_file[datastore]](https://github.com/ManageIQ/integration_tests/blob/48464ecdc292e119a87bde8678cd8363ce32782d/cfme/tests/automate/test_file_import.py#L55). There might be other tests as well, but I'm not aware of them.

Reply:
That test is failing to read the alert messages in the `AutomateImportExportView` class in `cfme/automate/import_export.py`, which already overrides the default `FlashMessages` class with the `V2VFlashMessages` subclass, and implements its own `ROOT` attribute:

```
class AutomateImportExportView(AutomateImportExportBaseView):
    @View.nested
    class flash(V2VFlashMessages):
        ROOT = 'div.import-flash-message'
```
It would make more sense to override `MSG_LOCATOR` here as well. This patch fixes the test in my manual testing:

```
diff --git a/cfme/automate/import_export.py b/cfme/automate/import_export.py
index 16cb23a13..618d8de05 100644
--- a/cfme/automate/import_export.py
+++ b/cfme/automate/import_export.py
@@ -42,6 +42,7 @@ class AutomateImportExportView(AutomateImportExportBaseView):
     @View.nested
     class flash(V2VFlashMessages):
         ROOT = 'div.import-flash-message'
+        MSG_LOCATOR = './div[contains(@class, "alert")]'
 
     @View.nested
     class import_file(View):  # noqa
```

The non-relative xpath locator `//` in your patch is effectively overriding `ROOT` and including it as part of your `MSG_LOCATOR`. They should be kept separate.

Reply:
@tpapaioa nice debug thanks. so its page specifc. I would say @dgaikwad close this PR and add one custom widget in `widgetastic_manageiq`.

I would not use `V2VFlashmessages` widget as it was solution for `TEXT_LOCATOR` change. As i can see `strong` tag in locator better just overwrite `Flashmessages` class ROOT and MSG_LOCATOR.

```
class ImportExportFlashMessages(FlashMessages):
    ROOT = ".//div[contains(@class, 'import-flash-message')]"
    MSG_LOCATOR = './/div[contains(@class, "alert")]'
```